datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyMulticastStrategy.java

```diff
@@ -205,7 +205,8 @@ public StickyMulticastStrategy(Optional<Integer> maxTasks, int imbalanceThreshol
       int minTasksPerInstance = tasksTotal / instances.size();
 
       // some rebalance to increase the task count in instances below the minTasksPerInstance
-      while (newAssignment.get(instancesBySize.get(0)).size() < minTasksPerInstance) {
+      while (newAssignment.get(instancesBySize.get(0)).size() + _imbalanceThreshold < newAssignment.get(
```

Comment:
This is a lot to parse in one line. Consider pulling out into a couple vars so the logic is more obvious.